src/Symfony/Component/HttpKernel/EventListener/SerializeResponseListener.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\SerializeResponse;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Automatically serializes controller return values when decorated with #[SerializeResponse].
+ */
+class SerializeResponseListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private SerializerInterface $serializer,
```

Comment:
the implementation should throw a useful error when the attribute is found but the serializer is null

```suggestion
        private ?SerializerInterface $serializer,
```